File: src/java/8/trunk/src/main/java/com/mycompany/Color.java
package com.mycompany;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Random;

/**
 * @author Paul Hammant DevOps, (c) 2018
 */
@JsonFormat(shape = JsonFormat.Shape.OBJECT)
enum Color {

  Blonde(),
  Brown(),
  Black(),
  Red();

  static int colorCtr = 0;

  Color() {
  }

  @JsonProperty()
  String getColor(){
    return name();
  }

  static Color rotatingChoice() {
    return Color.values()[colorCtr++ % Color.values().length];
  }

}

File: src/dotnet/8/trunk/Color.cs
using System.Text.Json.Serialization;

namespace MyCompany;

/// <author>Paul Hammant DevOps, (c) 2018</author>
[JsonConverter(typeof(JsonStringEnumConverter<Color>))]
public enum Color
{
    Blonde,
    Brown,
    Black,
    Red
}

public static class Colors
{
    private static int _colorCtr = 0;

    public static Color RotatingChoice()
    {
        var values = Enum.GetValues<Color>();
        return values[_colorCtr++ % values.Length];
    }
}
